packages/cozy-mespapiers-lib/src/components/Viewer/FileViewerWithQuery.jsx

```diff
@@ -36,7 +37,11 @@ const FilesViewerWithQuery = () => {
         fileId={currentFileId}
         files={filesQuery.data}
         filesQuery={filesQuery}
-        onClose={() => navigate(-1)}
+        onClose={() =>
+          navigate(
+            state?.background ? state.background : `/paper/files/${fileTheme}`
+          )
+        }
         onChange={fileId => navigate(`/paper/file/${fileId}`)}
       />
     </SharingProvider>
```

Comment:
pour le commit title c'est `Use state instead history on "navigate" ` plutôt non ?

Reply:
En effet, ce n'est pas très clair, mais ça serait plus quelque chose comme `Using paths rather than history to navigate`